### PR TITLE
Fixes #4443: Talkback reader reads blank on empty resume lesson summary

### LINKED ISSUE #4443: Talkback reader reads blank on empty resume lesson summary
**Describe the bug**
When we open the Ratios and Proportional reasoning story, the resume lesson summary is empty. So for such situation talkback reader is reading it blank instead of ignoring it. 

**To Reproduce**
1. Go to 'Ratios and Proportional Reasoning'
2. Click on any lesson
3. Turn on the screen reader
4. Navigate to all components on the screen using a screen reader. 

**Expected behavior**
Whenever the resume lesson summary is blank or empty, the talkback reader should ignore it instead of reading it out blank. 

**Demonstration**

https://user-images.githubusercontent.com/43074241/180382879-a4b364e2-92c0-4806-a476-86b3ddaf0bae.mp4

**Environment**
- Device being used: Realme GT Master Edition 
- Android or SDK version (e.g. Android 5 or SDK 21): Android 12
- App version (you can get this through system app settings or via the admin controls menu in-app): 1.0

### PR DESCRIPTION
**Pull request by vrajdesai78**
## Explanation
Fixes #4443: Handles edge case where the summary for resume lesson is empty. 

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

### Before 

https://user-images.githubusercontent.com/43074241/180387183-77931c08-c0d3-420a-b338-4dd694e4d9ef.mp4

### After

https://user-images.githubusercontent.com/43074241/180387741-9c198510-6a2f-4ec9-918d-7907bd03983c.mp4

### Passing Test cases
Screenshot from 2022-08-03 11-11-23
Screenshot from 2022-08-03 11-12-05

### CODE REVIEW

File: app/src/main/java/org/oppia/android/app/resumelesson/ResumeLessonFragmentPresenter.kt [hunk 1]
```diff
@@ -126,6 +126,10 @@ class ResumeLessonFragmentPresenter @Inject constructor(
       resumeLessonViewModel.chapterSummary.get()!!.summary,
       binding.resumeLessonChapterDescriptionTextView
     )
+    if (binding.resumeLessonChapterDescriptionTextView.text.toString().isBlank()) {
+      binding.resumeLessonChapterDescriptionTextView.importantForAccessibility =
+        View.IMPORTANT_FOR_ACCESSIBILITY_NO
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/resumelesson/ResumeLessonFragmentPresenter.kt]: @vrajdesai78 As per discussion we should add two tests (1.) for `View.IMPORTANT_FOR_ACCESSIBILITY_YES` and  (2.) for View.IMPORTANT_FOR_ACCESSIBILITY_NO

**Review by rt4914 — CHANGES_REQUESTED** [review]
@vrajdesai78 PTAL. Thanks.

**Comment by vrajdesai78** [comment]
Keep this PR open

**Comment by vrajdesai78** [comment]
When I tried to write test case https://gist.github.com/vrajdesai78/5088cd112ecf3c505ff13b79ac433bd1 it's showing me error as shown below. I have tried to create newTestChecking test to test whether textView is set to ImportantForAccessibility or Not. 

Screenshot from 2022-07-30 00-25-40

File: app/src/main/java/org/oppia/android/app/resumelesson/ResumeLessonFragmentPresenter.kt [hunk 2]
```diff
@@ -126,6 +126,10 @@ class ResumeLessonFragmentPresenter @Inject constructor(
...
+    }
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/resumelesson/ResumeLessonFragmentPresenter.kt:134]: @vrajdesai78 Suggest changing this entire logic like this:
```
  private fun updateChapterDescription() {
    val chapterDescription = htmlParserFactory.create(
      resourceBucketName,
      resumeLessonViewModel.entityType,
      explorationId,
      imageCenterAlign = true
    ).parseOppiaHtml(
      resumeLessonViewModel.chapterSummary.get()!!.summary,
      binding.resumeLessonChapterDescriptionTextView
    )
    if (chapterDescription.isNotBlank()) {
      binding.resumeLessonChapterDescriptionTextView.visibility = View.VISIBLE
      binding.resumeLessonChapterDescriptionTextView.text = chapterDescription
    } else {
      binding.resumeLessonChapterDescriptionTextView.visibility = View.GONE
    }
  }
```

**Why this change?**
In previous solution we are just controlling the accessibility-importance but if we enable `Show Layout Bounds` in `Developer Options` the `empty` textview is still available. But this solution gets rid of the `empty textview` completely which is much better solution.

I believe you can easily write test case for this solution?

**Review by rt4914 — CHANGES_REQUESTED** [review]
@vrajdesai78 PTAL.

File: app/src/main/java/org/oppia/android/app/resumelesson/ResumeLessonFragmentPresenter.kt (hunk 2 above)
Comment (vrajdesai78) [reply app/src/main/java/org/oppia/android/app/resumelesson/ResumeLessonFragmentPresenter.kt:134]: Yes, it sounds good to me. I have updated code based on your suggestion and also added test case for same. Thanks @rt4914

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks.

**Pull request merged**